src/cmd-archive

```diff
@@ -0,0 +1,749 @@
+        """
+        lookup = {
+            "commit": self.commit,
+            "config": self.config,
+            "image": self.image,
+            "meta": self.meta,
+        }
+        try:
+            return lookup[key]
+        except:
+            raise Exception("invalid key %s, valid keys are %s" %
+                            (key, lookup.keys()))
+
+    def get_meta_key(self, obj, key):
+        """
+        Look up a the key in a dict
+        @param dict: reference to the dictionary
+        @param str: key in dictionary to return
+        """
+        try:
+            data = self.get_obj(obj)
+            return data[key]
+        except KeyError as err:
+            log.warning("lookup for key '%s' returned: %s", key, str(err))
+            return None
+
+    def get_sub_obj(self, obj, key, sub):
+        """
+        Return the sub-element sub of key in a nested dic
+        @param dict: the parent dict
+        @param str: key of the sub element
+        @param str: the sub key to return
+        """
+        if isinstance(obj, str):
+            obj = self.get_obj(obj)
+            return self.get_sub_obj(obj, key, sub)
+        try:
+            return obj[key][sub]
+        except KeyError:
+            log.warning(obj)
+
+    def build_artifacts(self):
+        """
+        Walk the build root and prepare a list of files in it. While
+        commitmeta.json provides these artifacts, the information about the
+        individual files is not compatible with the Koji upload (e.g. SHA,
+        type has to be translated, strict filename conventions)
+        """
+        log.info("Processing the build artifacts")
+
+        # locate all the build artifacts
+        files = []
+        for ffile in os.listdir(self.build_root):
+            files.append(os.path.join(self.build_root, ffile))
+            if os.path.islink(ffile):
+                log.debug(" * EXCLUDING symlink '%s'", ffile)
+                log.debug(" *    target '%s'", os.path.realpath(ffile))
+                continue
+
+        # add the coreos assembler files
+        for ffile in os.listdir(COSA_INPATH):
+            files.append(os.path.join(COSA_INPATH, ffile))
+
+        # process the files that were found
+        for ffile in files:
+            log.debug("Considering file file '%s'", ffile)
+            short_path = os.path.basename(ffile)
+
+            if not koji_upload(short_path):
+                log.debug(" * EXCLUDING file '%s'", ffile)
+                log.debug("   File type is not supported by Koji")
+                continue
+
+            # os.path.getsize returns an incorrect value, so use stat
+            fsize = subprocess.check_output(["stat", "--format", '%s', ffile])
+            log.debug(" * calculating checksum")
+            self._found_files[ffile] = {
+                "local_path": os.path.abspath(ffile),
+                "path": short_path,
+                "md5": md5sum_file(ffile),
+                "size": int(fsize)
```

Comment:
nit:  other `dicts` defined in this code have the trailing `,` on the last element, I'd just continue the trend here

Reply:
Fixed up